soft4pes/sim/simulation.py

```diff
@@ -51,43 +101,38 @@ def simulate(self, t_stop):
             Simulation stop time [s]. Simulation start time is always 0 s.
         """
 
-        # For saving data, should be done in a better way
+        progress_printer = ProgressPrinter(int(t_stop / self.ctr.Ts))
+
         data = np.empty((0, 2))
         time = np.empty((0, 1))
         u_data = np.empty((0, 3))
         self.t_stop = t_stop
 
         t = 0
-        # This expects that t_stop/Ts is an integer
         for i in range(int(self.t_stop / self.ctr.Ts)):
 
             # Execute the controller
             u = self.ctr(self.sys, self.conv, t)
 
-            # Save data, for debugging and verification, will be removed later
-            data = np.vstack((data, self.sys.x))
-            time = np.vstack((time, t))
-            u_data = np.vstack((u_data, u))
-
-            matrices = self.sys.get_discrete_state_space(
-                self.conv.v_dc, self.ctr.Ts)
-            vg = self.sys.get_grid_voltage(t)
+            for _ in range(int(self.ctr.Ts / self.Ts)):
 
-            x_kp1 = np.dot(matrices.A, self.sys.x) + np.dot(
-                matrices.B1, u) + np.dot(matrices.B2, vg)
+                # Save data
+                data = np.vstack((data, self.sys.x))
+                time = np.vstack((time, t))
+                u_data = np.vstack((u_data, u))
 
-            self.sys.update_state(x_kp1)
+                self.sys.update_state(u, self.matrices, t)
 
-            t = t + self.ctr.Ts
+                t = t + self.Ts
 
-            print("{}%".format(np.round(i / (t_stop / self.ctr.Ts) * 100, 1)))
+            progress_printer(i)
 
-        # Plot data, debugging
+        # Plot data
         plt.plot(time, data)
```

Comment:
I think it would be better to define a new separate function for plot settings, such as fonts, x and y axis titles, legends, etc. (to show results in a better way)

Reply:
I agree. This will be a big future task that requires thinking and creating the whole plotting system from the scratch, so I will not implement that yet.